src/components/institution/AffiliationHierarchy.tsx

```diff
@@ -1,25 +1,40 @@
-import React, { FC } from 'react';
-import { RecursiveInstitutionUnit } from '../../types/institution.types';
+import React, { FC, Fragment } from 'react';
+import { CircularProgress } from '@material-ui/core';
+
 import Label from '../Label';
 import NormalText from '../NormalText';
+import useFetchUnitHierarchy from '../../utils/hooks/useFetchUnitHierarchy';
+import { getUnitHierarchyNames } from '../../utils/institutions-helpers';
 
 interface AffiliationHierarchyProps {
-  unit: RecursiveInstitutionUnit;
-  boldTopLevel?: boolean;
+  unitUri: string;
+  commaSeparated?: boolean; // Comma separated or line breaks
+  boldTopLevel?: boolean; // Only relevant if commaSeparated=true
 }
 
-const AffiliationHierarchy: FC<AffiliationHierarchyProps> = ({ unit, boldTopLevel = true }) => (
-  <>
-    {boldTopLevel ? <Label>{unit.name}</Label> : <NormalText>{unit.name}</NormalText>}
-    {unit?.subunits && <SubUnitRow unit={unit.subunits[0]} />}
-  </>
-);
+export const AffiliationHierarchy: FC<AffiliationHierarchyProps> = ({
+  unitUri,
+  commaSeparated = false,
+  boldTopLevel = true,
+}) => {
+  const [unit, isLoadingUnit] = useFetchUnitHierarchy(unitUri);
+  const unitNames = unit ? getUnitHierarchyNames(unit) : [];
 
-const SubUnitRow: FC<AffiliationHierarchyProps> = ({ unit }) => (
-  <>
-    <NormalText>{unit.name}</NormalText>
-    {unit.subunits && <SubUnitRow unit={unit.subunits[0]} />}
-  </>
-);
+  return isLoadingUnit ? (
+    <CircularProgress size={20} />
+  ) : unit ? (
+    commaSeparated ? (
+      <NormalText>{unitNames.join(', ')}</NormalText>
+    ) : (
+      <div>
```

Comment:
må du ha divs her? kunne kanskje flyttet de til den inni map'en isteden for fragment?

Reply:
ville egentlig ha det, slik at vi vet at komponenten returnerer 1 tag, som kan stiles som vi vil. 
eks: i contributors har vi <AffiliationsHierarchy /> i en flex-box som ellers ville gjort at alt plasseres etter hverandre horisontalt. så enten må alt legges i en div her, eller så kan vi legge inn en div hvor det faktisk trengs, og bare bruke `<>...</>` her. hva liker du best?

Reply:
åja, sånn ja! nei, men da er det vel ok 👍